assets/js/modules/ads/components/common/PAXEmbeddedApp.js

```diff
@@ -149,6 +162,26 @@ export default function PAXEmbeddedApp( {
 		launchPAXApp,
 	] );
 
+	/* eslint-disable react-hooks/exhaustive-deps */
+	// Rule disabled for compiled `paxDateRange` properties, instead of using the main object
+	// to ensure effect runs every time dates are changed.
+	useEffect( () => {
+		if ( displayMode === 'reporting' && paxAppRef?.current ) {
+			const { startDate, endDate } = paxDateRange;
+
+			paxAppRef.current.getServices().adsDateRangeService.update( {
+				startDate: formatPaxDate( startDate ),
+				endDate: formatPaxDate( endDate ),
+			} );
+		}
+	}, [
+		paxAppRef?.current,
+		registry,
+		displayMode,
+		`${ paxDateRange?.startDate }|${ paxDateRange?.endDate }`,
+	] );
+	/* eslint-enable react-hooks/exhaustive-deps */
```

Comment:
Have you tried destructuring `paxDateRange` above like so:

```
   const { current, startDate, endDate } = useSelect( ... ) || {}
```

Could be cleaner and remove the need disabling the rule, although this may not fix the issue with updating correctly with the date updates.

Reply:
The idea is to source data outside the `useEffect` as dates are needed as dependency. So we need to keep it there. Eslint rule is due to the not using variable directly but as parsed string.

Reply:
Won't creating the string as a new object cause this to re-run on _every_ render? Either way, removing the hook warning about exhaustive dependencies means other errors could be made here, and ignoring these hook rules can lead to subtle bugs. I'd rather another way around this. 🤔